### PR TITLE
feat(Raven Code): buttons, docs subdomain

### PR DESCRIPTION
**Pull request by chimpdev**
## Description 
- Adding buttons
- Support for docs subdomain
- Bumping version

## Acknowledgements
- [X] I read the [Presence Guidelines](https://github.com/PreMiD/Presences/blob/main/.github/CONTRIBUTING.md)
- [X] I linted the code by running `yarn format`
- [X] The PR title follows the repo's [commit conventions](https://github.com/PreMiD/Presences/blob/main/.github/COMMIT_CONVENTION.md)

## Screenshots
<details>
<summary> Proof showing the creation/modification is working as expected </summary>

image
image
image
(old logo asset already changed on portal but not updated, probably cache issue)

</details>

### CODE REVIEW

**Comment by chimpdev** [comment]
sorry for 2st pr 😑

**Review by DarkVillager — CHANGES_REQUESTED** [review]

**Comment by DarkVillager** [comment]
@chimpdev mark changes that you’ve done as resolved

File: websites/R/Raven Code/presence.ts [hunk 1]
```diff
@@ -14,74 +14,81 @@ presence.on("UpdateData", async () => {
...
-						"[data-premid-usertag]"
-					)?.innerText;
-				if (image && userTag) {
-					presenceData.details = "Kullanıcıyı görüntülüyor:";
-					presenceData.state = userTag;
-					presenceData.largeImageKey = image;
+	if (host === "docs.ravencode.live") {
+		presenceData.details = "docs.ravencode.live";
+		presenceData.state = document.title.replaceAll("-", "—");
+	} else if (host === "ravencode.live") {
```
Comment (DarkVillager) [inline websites/R/Raven Code/presence.ts:26]: Optional but also change this to a switch statement for readability sake?

File: websites/R/Raven Code/presence.ts [hunk 2]
```diff
@@ -14,74 +14,81 @@ presence.on("UpdateData", async () => {
...
+						presenceData.state = userTag;
+						presenceData.largeImageKey = image;
+					}
+				} else if (searchParams.has("invite_code")) {
+					const image = document.querySelector<HTMLImageElement>(
+							"[data-premid-guildicon]"
+						)?.src,
+						guildName = document.querySelector<HTMLElement>(
+							"[data-premid-guildname]"
+						)?.innerText;
```
Comment (DarkVillager) [inline websites/R/Raven Code/presence.ts:48]:
```suggestion
						)?.textContent;
```

File: websites/R/Raven Code/presence.ts [hunk 3]
```diff
@@ -14,74 +14,81 @@ presence.on("UpdateData", async () => {
...
+		switch (pathname) {
+			case "/":
+				presenceData.details = "Anasayfayı görüntülüyor";
+				if (searchParams.has("user_id")) {
+					const image = document.querySelector<HTMLImageElement>(
+							"[data-premid-useravatar]"
+						)?.src,
+						userTag = document.querySelector<HTMLElement>(
+							"[data-premid-usertag]"
+						)?.innerText;
```
Comment (DarkVillager) [inline websites/R/Raven Code/presence.ts:36]:
```suggestion
						)?.textContent;
```

File: websites/R/Raven Code/presence.ts [hunk 4]
```diff
@@ -14,74 +14,81 @@ presence.on("UpdateData", async () => {
...
+					if (image && userTag) {
+						presenceData.details = "Kullanıcıyı görüntülüyor:";
+						presenceData.state = userTag;
```
Comment (DarkVillager) [inline websites/R/Raven Code/presence.ts:39]: This could be undefined due to the  check at the top.
```suggestion
						presenceData.state = userTag ?? "logo";
```

**Review by DarkVillager — CHANGES_REQUESTED** [review]

File: websites/R/Raven Code/presence.ts (hunk 4 above)
Comment (chimpdev) [reply websites/R/Raven Code/presence.ts:39]: This is never undefined.

File: websites/R/Raven Code/presence.ts (hunk 1 above)
Comment (chimpdev) [reply websites/R/Raven Code/presence.ts:26]: not necessary

**Comment by EncryptedDev** [comment]
Resolve CI

File: websites/R/Raven Code/dist/metadata.json [hunk 5]
```diff
@@ -9,15 +9,11 @@
...
 	"thumbnail": "https://i.imgur.com/LabRRNC.png",
-	"color": "#121316",
+	"color": "#7a92f1",
 	"category": "other",
-	"tags": [
-		"discord",
-		"kod-paylaşım",
-		"raven-code"
-	]
+	"tags": ["discord", "kod-paylaşım", "raven-code"]
```
Comment (EncryptedDev) [inline websites/R/Raven Code/dist/metadata.json:18]: I believe the former version was preferable hence why we had it - revert this change

File: websites/R/Raven Code/presence.ts [hunk 6]
```diff
@@ -14,74 +14,81 @@ presence.on("UpdateData", async () => {
 	const presenceData: PresenceData = {
 			largeImageKey: "logo",
 			details: "Geziniyor",
+			buttons: [{ label: "Sayfaya git", url: document.location.href }],
```
Comment (EncryptedDev) [inline websites/R/Raven Code/presence.ts:17]: This is bad practice

File: websites/R/Raven Code/presence.ts [hunk 7]
```diff
@@ -14,74 +14,81 @@ presence.on("UpdateData", async () => {
...
-					)?.src,
-					userTag = document.querySelector<HTMLElement>(
-						"[data-premid-usertag]"
-					)?.innerText;
-				if (image && userTag) {
-					presenceData.details = "Kullanıcıyı görüntülüyor:";
-					presenceData.state = userTag;
-					presenceData.largeImageKey = image;
+	if (host === "docs.ravencode.live") {
+		presenceData.details = "docs.ravencode.live";
```
Comment (EncryptedDev) [inline websites/R/Raven Code/presence.ts:24]: This isn't descritive at all, use the value from state here instead and remove details

**Review by EncryptedDev — CHANGES_REQUESTED** [review]

File: websites/R/Raven Code/presence.ts (hunk 6 above)
Comment (chimpdev) [reply websites/R/Raven Code/presence.ts:17]: why?

File: websites/R/Raven Code/dist/metadata.json (hunk 5 above)
Comment (chimpdev) [reply websites/R/Raven Code/dist/metadata.json:18]: I did not understand

Comment (EncryptedDev) [reply websites/R/Raven Code/dist/metadata.json:18]: Rever the change on this line

File: websites/R/Raven Code/presence.ts (hunk 6 above)
Comment (EncryptedDev) [reply websites/R/Raven Code/presence.ts:17]: You should have buttons specifically for each page where appropriate, not a "Go to page" button for every page

**Comment by EncryptedDev** [comment]
Respond to my reviews

File: websites/R/Raven Code/presence.ts (hunk 6 above)
Comment (chimpdev) [reply websites/R/Raven Code/presence.ts:17]: ok, why?

Comment (RisingSunLight42) [reply websites/R/Raven Code/presence.ts:17]: > ok, why?

Because that's low quality and too generic.

File: websites/R/Raven Code/dist/metadata.json (hunk 5 above)
Comment (chimpdev) [reply websites/R/Raven Code/dist/metadata.json:18]: I didn't do it. Prettier did it.

File: websites/R/Raven Code/presence.ts (hunk 6 above)
Comment (chimpdev) [reply websites/R/Raven Code/presence.ts:17]: That's how I want it to be.

Comment (theusaf) [reply websites/R/Raven Code/presence.ts:17]: However, it seems that this is not how PreMiD wants it to be.

File: websites/R/Raven Code/dist/metadata.json (hunk 5 above)
Comment (theusaf) [reply websites/R/Raven Code/dist/metadata.json:18]: Run `yarn format` or `yarn ms` to reformat this file or revert the change manually. Prettier does mess with the file if you are not careful, and it may be a good idea to only prettify your ts files or undo changes prettier does to the metadata files after running prettier.

File: websites/R/Raven Code/presence.ts (hunk 6 above)
Comment (chimpdev) [reply websites/R/Raven Code/presence.ts:17]: Okay.

**Comment by chimpdev** [comment]
Pr can close
